--- deluge/ui/gtk3/gtkui.py.orig	2019-06-12 17:49:45.000000000 +0000
+++ deluge/ui/gtk3/gtkui.py	2019-06-15 18:18:50.000000000 +0000
@@ -155,15 +155,15 @@
 
             SetConsoleCtrlHandler(on_die, True)
             log.debug('Win32 "die" handler registered')
-        elif osx_check() and windowing('quartz'):
-            try:
-                import gtkosx_application
-            except ImportError:
-                pass
-            else:
-                self.osxapp = gtkosx_application.gtkosx_application_get()
-                self.osxapp.connect('NSApplicationWillTerminate', on_die)
-                log.debug('OSX quartz "die" handler registered')
+#       elif osx_check() and windowing('quartz'):
+#           try:
+#               import gtkosx_application
+#           except ImportError:
+#               pass
+#           else:
+#               self.osxapp = gtkosx_application.gtkosx_application_get()
+#               self.osxapp.connect('NSApplicationWillTerminate', on_die)
+#               log.debug('OSX quartz "die" handler registered')
 
         # Set process name again to fix gtk issue
         setproctitle(getproctitle())
